cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,260 @@
 -   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
+
+    A Kubernetes [admission controller webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+    which can be used to dynamically convert logical resource names in pod specifications into actual FPGA
+    resource names, as advertised by the device plugin.
+
+    The webhook can also set environment variables to instruct the FPGA prestart hook to program the FPGA
+    before launching the container.
+
 -   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
 
+    A [CRI-O](https://github.com/cri-o/cri-o) prestart hook that, upon instruction from the FPGA admission
+    controller, allocates and programs the FPGA before the container is launched.
 
-### Get source code:
-```
-    $ mkdir -p $GOPATH/src/github.com/intel/
-    $ cd $GOPATH/src/github.com/intel/
-    $ git clone https://github.com/intel/intel-device-plugins-for-kubernetes.git
-```
+The repository also contains an [FPGA helper tool](../fpga_tool/README.md) that may be useful during
+development, initial deployment and debugging.
 
-### Build FPGA device plugin:
-```
-    $ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-    $ make fpga_plugin
-```
+# Installation
+
+The below sections cover how to obtain, build and install this component.
+
+Components can generally be installed either using DaemonSets or running them
+'by hand' on each node.
+
+## Pre-built images
+
+Pre-built images of the components are available on the [Docker hub](https://hub.docker.com/u/intel).
+These images are automatically built and uploaded from the latest `master` branch of this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers.
+
+The deployment YAML files supplied with these components in this repository use the images with the
+`devel` tag by default. If you do not build your own local images, then your Kubernetes cluster may
+pull down the `devel` images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the YAML deployment files appropriately.
+
+The following images are available on the Docker hub:
+
+- [The FPGA plugin](https://hub.docker.com/r/intel/intel-fpga-plugin)
+- [The FPGA admisson webhook](https://hub.docker.com/r/intel/intel-fpga-admissionwebhook)
+- [The FPGA CRI-O plugin (in the `initcontainer` image)](https://hub.docker.com/r/intel/intel-fpga-initcontainer)
+
+## Dependencies
+
+All components have the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about)
+
+To obtain a fully operational FPGA enabled cluster, you must install all three
+major components:
+
+-   [FPGA device plugin](README.md) (this component)
+-   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
+-   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
+
```

Comment:
It makes sense to emphasize that plugin depends on CRI-O runtime in orchestration programmed mode only. It can use any other runtime in preprogrammed mode.

Reply:
FPGA plugin also depends on Linux Kernel device drivers. It supports 2 drivers:

- [DFL](https://www.kernel.org/doc/html/latest/fpga/dfl.html) in-tree driver
- [OPAE](https://opae.github.io/latest/docs/drv_arch/drv_arch.html) out of tree driver

Reply:
Done. Thanks for the links. added and updated.